### PR TITLE
Added switch for repeat purchase check

### PR DESCRIPTION
**Pull request by clintonb**
This check to avoid allowing learners purchasing the same item multiple times is a nuisance in development and test environments. If refund deadlines pass, there is no simple way for developers to get into a state where they can view the checkout page. Disabling this check allows for the addition of items regardless of whether they have been previously purchased.

### CODE REVIEW

**Comment by clintonb** [comment]
@edx/learner please review

**Review by mjfrey — APPROVED** [review]
I still am not a fan of the check in the first place.

**Review by schenedx — APPROVED** [review]
👍

**Comment by schenedx** [comment]
@Ayub-Khan Please be aware of this change

**Comment by Ayub-Khan** [comment]
@clintonb  @schenedx @mjfrey I Just had a look at this pull request . This pull request should not have been merged. Current implementation of 'allow_repeat_purchase' is doing the exact opposite job as compared to its name. If this flag is disabled It will work as a kill switch for all the basket views.

**Comment by clintonb** [comment]
The switch is misnamed. It should be called disallow_repeat_purchase. The logic itself is correct. I'll update the name later today.

File: ecommerce/extensions/basket/utils.py [hunk 1]
```diff
@@ -43,24 +45,26 @@ def prepare_basket(request, products, voucher=None):
...
-            )
-    if already_purchased_products and basket.is_empty:
-        raise AlreadyPlacedOrderException
+
+    if waffle.switch_is_active(REPEAT_PURCHASE_SWITCH_NAME):
+        already_purchased_products = []
+        for product in products:
+            if product.is_enrollment_code_product or \
+                    not UserAlreadyPlacedOrder.user_already_placed_order(request.user, product):
+                basket.add_product(product, 1)
```
Comment (schenedx) [inline ecommerce/extensions/basket/utils.py:54]: @clintonb If the switch is not active, according to the change here, where do we add the products to the basket?
Doesn't this mean if the switch is not active, we are not adding ANY product to the basket?

Comment (clintonb) [reply ecommerce/extensions/basket/utils.py:54]: Hmm...I don't have time to properly debug this. I will revert the commit.

Comment (clintonb) [reply ecommerce/extensions/basket/utils.py:54]: Correction: my initial statement stands. We should not deactivate this switch. I'll fix it later today.

Comment (Ayub-Khan) [reply ecommerce/extensions/basket/utils.py:54]: @clintonb no need to revert this. its working fine unless the switch is deactivated. another quick question was this switch in off state yesterday for some time ?.

File: ecommerce/extensions/basket/utils.py [hunk 2]
```diff
@@ -43,24 +45,26 @@ def prepare_basket(request, products, voucher=None):
...
-                mode_for_seat(product),
-                product.course_id
-            )
-    if already_purchased_products and basket.is_empty:
-        raise AlreadyPlacedOrderException
+
+    if waffle.switch_is_active(REPEAT_PURCHASE_SWITCH_NAME):
+        already_purchased_products = []
+        for product in products:
+            if product.is_enrollment_code_product or \
```
Comment (Ayub-Khan) [inline ecommerce/extensions/basket/utils.py:52]: I think adding switch here as 'or' will fix the pr.

**Comment by clintonb** [comment]
I created https://github.com/edx/ecommerce/pull/1339 to fix the identified issues.

**Pull request merged**